processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -214,7 +214,26 @@ public File mergeQueryableIndex(
     return mergeQueryableIndex(indexes, rollup, metricAggs, outDir, indexSpec, new BaseProgressIndicator());
   }
 
-  public File mergeQueryableIndex(
+  public File mergeAndCloseMergee(
+      List<QueryableIndex> indexes,
+      boolean rollup,
+      final AggregatorFactory[] metricAggs,
+      File outDir,
+      IndexSpec indexSpec,
+      ProgressIndicator progress
+  ) throws IOException
+  {
+    try {
+      return mergeQueryableIndex(indexes, rollup, metricAggs, outDir, indexSpec, progress);
+    }
+    finally {
```

Comment:
My first intension was using that but it made change much bigger. In my experience, it'll make PRs hard to be committed. If any committer grants me, I'll do that.